app/services/compute_resource_host_importer.rb

```diff
@@ -0,0 +1,69 @@
+class ComputeResourceHostImporter
+  attr_accessor :host, :compute_resource, :vm
+
+  delegate :logger, :to => :Rails
+
+  def initialize(opts = {})
+    self.compute_resource = opts.fetch(:compute_resource)
+    self.vm = compute_resource.find_vm_by_uuid(opts[:uuid]) if opts[:uuid]
+    self.vm ||= opts.fetch(:vm)
```

Comment:
`find_vm_by_uuid` raises `ActiveRecord::NotFound` so we should be covered. Your suggestion is fine as well, though.